src/meetingreadinesschecker/DefaultMeetingReadinessChecker.ts

```diff
@@ -270,7 +270,25 @@ export default class DefaultMeetingReadinessChecker implements MeetingReadinessC
 
   async checkAudioConnectivity(audioInputDevice: Device): Promise<CheckAudioConnectivityFeedback> {
     let audioPresence = false;
+    const audioConnectivityMetrics = {
+      packetsSent: 0,
+      packetsReceived: 0,
+    };
     const audioVideo = this.meetingSession.audioVideo;
+
+    const checkAudioConnectivityMetricsObserver: AudioVideoObserver = {
+      metricsDidReceive(clientMetricReport: ClientMetricReport) {
+        clientMetricReport.getRTCStatsReport().forEach(report => {
+          if (report.type === 'outbound-rtp' && report.mediaType === 'audio') {
+            audioConnectivityMetrics.packetsSent = report.packetsSent;
```

Comment:
Hmm why are we checking packets sent here as part of connectivity check? The idea is to allow an attendee to join the call even if they are not sending any audio packets. So a check for packets being received, in addition to the presence, should be enough.

Reply:
Please correct me if I am wrong here. My understanding is that - Prior to the fast join, attendee presence used to indicate that the audio packet is being sent and received by the attendee. So, as part of *AudioConnectivity* check in meetingReadinessChecker, we want to check whether the attendee is able to send and receive the audio packets. 
Now, after fast join, audio packets sent and received in addition to attendee presence should account for audio connectivity check. 
And if we are to show the connectivity where attendee can join the call even if they are not sending any audio packets, we should show a separare connectivity check - Say *AudioReceiveConnectivity* check inside meetingReadinessChecker. What do you think ?

Reply:
Currently, with fast join not enabled, attendee presence message is sent out by the server on receiving an audio packet from the client. So the connectivity check today, checks one way audio. 

When we deploy fast join, presence message will be received by client on the signaling channel without it needing to send out an audio packet. But to ensure the clients media channel is connected, we are adding the check for received packets. 

Sending of packets is very dependent on the audio device being used. I think sending packet health should be a separate check and audio connectivity check should include the presence and received audio check.

Reply:
Thanks for the clarification.

Reply:
I agree with Pracheth's thought process here,but given that [`checkVideoConnectivity`](https://github.com/aws/amazon-chime-sdk-js/blob/553c2d2f8f18abd2c3daef487ac5e4173afa93c6/src/meetingreadinesschecker/DefaultMeetingReadinessChecker.ts#L312-L353)  uses packetsSent to determine success and returns a VideoNotSent if it fails, I'm inclined to just keep a single connectivity check and add AudioNotSent to the CheckAudioConnectivityFeedback enum and return that if packetsSent is not greater than zero. Thoughts?

Reply:
Agree, packetsSent metric in addition to attendee Presence can indicate audio connectivity, plus the packetsSent metric is consistent with the current checkVideoConnectivity as well.

Reply:
I agree "Connectivity" is not clearly defined here. But, I think by checking all three things, we'll cover all grounds and your rationale makes sense to me.

Reply:
What if a listen only attendee is using the readiness checker before joining the meeting? Given that we won't require builders to have a "fake" audio stream sending muted packets soon, this attendees readiness check will fail as they will never send out packets.

Reply:
Previously, rather currently, the `checkAudioConnectivity` function only checked for presence and given that the server waits for a mic packet to send that out today, it was technically on testing the sending of audio. We weren't checking for packets being received anywhere. Note that the presence message is received by the client on the signaling channel.

Reply:
Updated the checkAudioConnectivity logic as per the suggestions. Thanks